battles/helpers.py

```diff
@@ -25,21 +27,53 @@ def get_one_a_one_fight_winner(pokemon_1, pokemon_2):
 
 
 # battle fight
-def get_battle_winner(pokemons_creator, pokemons_opponent):
+def get_battle_winner(battle):
+    trainer_team_creator = TrainerTeam.objects.get(
+        Q(battle_related=battle), Q(trainer=battle.trainer_creator)
+    )
+    trainer_team_opponent = TrainerTeam.objects.get(
+        Q(battle_related=battle), Q(trainer=battle.trainer_opponent)
+    )
+    pokemons_creator = [
+        trainer_team_creator.pokemon_1,
+        trainer_team_creator.pokemon_2,
+        trainer_team_creator.pokemon_3
+    ]
+    pokemons_opponent = [
+        trainer_team_opponent.pokemon_1,
+        trainer_team_opponent.pokemon_2,
+        trainer_team_opponent.pokemon_3,
+    ]
     one_a_one_results = []
+
     for (pokemon_c, pokemon_o) in zip(pokemons_creator, pokemons_opponent):
         if get_one_a_one_fight_winner(pokemon_c, pokemon_o) == pokemon_c:
             one_a_one_results.append('creator')
         else:
             one_a_one_results.append('opponent')
 
     if one_a_one_results.count('creator') > one_a_one_results.count('opponent'):
-        return 'creator'
-    return 'opponent'
+        return battle.trainer_creator
+    return battle.trainer_opponent
+
+
+def run_battle(battle):
+    battle.status = 'SETTLED'
+    battle.trainer_winner = get_battle_winner(battle)
+    return battle.save()
+
+
+def order_battle_pokemons(cleaned_data):
+    pokemons = [0, 1, 2]
```

Comment:
@lucabezerra I did that way because I needed to insert the items in a specific index, like I do here:
```
pokemons[int(cleaned_data['order_1'])] = cleaned_data['pokemon_1'].name
```
Is there a way to a insert items in a specific index in the list without pre-allocating space in it?

Reply:
Sorry, didn't get notified about this comment!

So, yeah, if you add them directly like I put above, they'll remain in that order. Another option would be to `.append()` one after the other (they get appended to the end of the list), but since you already have all items at once, it's better to just create the list with them in it.

Reply:
I hadn't understood your question correctly at first, haha. 
So, just to be clear, we usually don't pre-instantiate lists in Python, because they allocate memory dynamically, as you probably know :)
Just for teaching an alternative, a more pythonic way of doing this would be (I haven't run the code, might be broken):
``` python
    dict_values = [{key, value} for key, value in cleaned_data.items() if key.startswith('order_')]
    pokemons = []
    for key, value in dict_values:
      pokemons.append({
        'order': value,
        'pokemon': cleaned_date.get('pokemon_' + key.split('_')[1])
      })
      
    pokemons = [item['pokemon'].name for item in sorted(pokemons, key=lambda x: x['order'])]
```
This method would even allow you to add as many pokemons as you want, instead of being limited to 3. However, as you probably noticed, this requires a lot of overhead code. For your specific need, I think your approach might actually be better, although pretty unconventional for Python, haha. Let's leave it as it is, then :)